### DISCUSSION TITLE
Phase B angle 255

### DISCUSSION THREAD

**Discussion by farcouet** [emporia-vue-local/esphome#238] [Q&A]
Hello,
I just flashed one of my Emporia to start (I will flash shortly the second one as I am sick of the unreliable Emporia Cloud that is down again).
1- Voltage calibration: When I read the voltage there was a big difference between phase A and B. Now I am using those values:
Phase A calibration: 0.0095
Phase B calibration: 0.014
I updated the calibration and I was getting around 122V. After few minutes now it's showing a quite big difference between my Fluke measure (122V on each phase with 0.2V variation) as it show currently 
Phase A 124.8 V / Phase B 133.2V

2- I see Phase B angle of 255 degres which makes no sense. For a quick test without taking out the Fluke Scopemeter I measured 244V between both phase with the Fluke true rms meater. So that confirms that I should be at 180 degrees (it's single phase in Canada normally in residential so I have two opposite phase).
I am not sure why and if this will impact power measure. 

I just want to be sure before flashing the main Emporia ;-)

**Comment by flaviut** [comment]
> Phase A calibration: 0.0095
> Phase B calibration: 0.014

If I recall correctly, the reasonable range here is 0.020 to 0.033. Outside this range indicates there's something else wrong here.

> I see Phase B angle of 255 degres which makes no sense

Yup... this seems like it might be the max value for a byte, but it's just a coincidence. The phase angle is a 16-bit integer.

Do you have the correct color/phase pairing in your config?

**Reply by farcouet** [reply]
Hello,
I will do mode test but yes for Phase A it’s the black, Phase B is the red and the blue is connected to neutral (when not used they were saying to put to neurtal to avoid false reading) as the white.
Clamps matches.
So this is strange because I did not had issues with it normally apart when their crappy cloud is down :-)
Let me know if any other ideas. I guess that next step will be to either flash it back and check again or put the logic analyzer and try to see what it’s reading.
Cheers,
Fred

**Reply by flaviut** [reply]
logic analyzer is great, but not needed here. Try something like

```
logger:
  level: DEBUG
```

and (from memory) it should dump the i2c data to the console. [You can then clone this spreadsheet to manually decode it.](https://docs.google.com/spreadsheets/d/1Yu98tDRhFJqqMntNak6X297SKyzQtf0_Cvv_g7qKewk/edit#gid=0)

>  So this is strange because I did not had issues with it normally apart when their crappy cloud is down :-)

Emporia automatically corrects for many common errors and misconfigurations. They are right to do this, even if it leads to slight inaccuracy, since their customers don't care and will return it if there's any trouble at all. This project doesn't do that because it's more work and, to a lesser degree, philosophically unsatisfying to me.

**Reply by farcouet** [reply]
Hello, 
I will do the i2c data dump but before I did few tests.
I am on 120/240V so using black and red and clamp A and B for the main monitoring. So the blue wire is connected to neutral as per the instructions. I tried to move it with the red and with the black and was able to see amps recorded on the B phase. But without this I was getting nothing and still the angle totally wrong (should be 180). when changing the position of the Blue wire then I also see the phase changing (which makes no sense) and it's changing also the voltages on A and B! 

I then just tried to flash it back with the original firmware and now I record the exact same amps on Black and Red (I used my Fluke clamp to compare with real and I was just using a heater so purely resistive). Everything matches with stock firmware.

So I really don't know what could I be doing wrong. Is there a way to record amps instead of power? Any other idea?
Cheers,
Fred
